include/seqan3/argument_parser/detail/concept.hpp

```diff
@@ -0,0 +1,53 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2021, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2021, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ * \brief Provides the concept seqan3::detail::is_list_option.
+ */
+
+#pragma once
+
+#include <string>
+#include <seqan3/std/type_traits>
+
+#include <seqan3/core/platform.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\interface seqan3::detail::is_list_option <>
+ * \ingroup argument_parser
+ * \brief Decides whether an option type is considered a list.
+ * \details
+ *
+ * In the seqan3::argument_parser::add_option or seqan3::argument_parser::add_positional_argument calls,
+ * it is important whether the type of the value, in which the respective option is stored (option_type),
+ * is a list or not because nternal handling of option lists is different.
+ *
+ * This concept decides whether an option type is considered a list or not.
+ * In general, all standard library containers except std::string can be considered option lists.
+ *
+ * In order to be considered an option list, the `option_type` as to fulfil the following:
+ * * `option_type` may not be `std::string`
+ * * Member type `option_type::value_type` must exist
+ * * Member function `option_type::push_back(typename option_type::value_type) -> void` must exist.
+ *
+ * \noapi{Exposition only.}
+ */
+//!\cond
+template <typename option_type>
```

Comment:
Would this be easier to read?

```cpp
template <typename option_type>
SEQAN3_CONCEPT is_list_option = !std::same_as<std::remove_cvref_t<option_type>, std::string> &&
                                requires (option_type option)
{
    typename std::remove_reference_t<option_type>::value_type;

    requires requires (std::remove_reference_t<option_type>::value_type value)
    {
        SEQAN3_RETURN_TYPE_CONSTRAINT(option.push_back(value), std::same_as, void);
    };
};
```

You could also shorten it to

```cpp
template <typename option_type>
SEQAN3_CONCEPT is_list_option = !std::same_as<std::remove_cvref_t<option_type>, std::string> &&
                                requires (option_type option, std::remove_reference_t<option_type>::value_type value)
{
    SEQAN3_RETURN_TYPE_CONSTRAINT(option.push_back(value), std::same_as, void);
};
```

the compiler would still tell you that it cannot find `value_type` for some `option_type`.
I don't think we ever discussed this, but it might be enough to do the `typename` constraints iff you don't need them anywhere else in the concept.

Reply:
Ah I see. I wondered about that but thought the comiler message is better that way. Thakns for clarifying :)